borg/upgrader.py

```diff
@@ -33,9 +37,19 @@ def upgrade(self, dryrun=True, inplace=False):
             logger.info('making a hardlink copy in %s', backup)
             if not dryrun:
                 shutil.copytree(self.path, backup, copy_function=os.link)
+                # we need to create a real copy (not hardlink copy) of index.* and hints.*
+                # otherwise the backup copy will get modified.
+                transaction_id = self.get_index_transaction_id()
+                for name in ['index', 'hints']:
+                    fname = "%s.%d" % (name, transaction_id)
+                    fname_orig = os.path.join(self.path, fname)
+                    fname_backup = os.path.join(backup, fname)
+                    os.remove(fname_backup)
+                    shutil.copy(fname_orig, fname_backup)
```

Comment:
i believe this is incorrect: the problem was that we were doing inplace header changes on the index files later, not that we were not copying them. this introduces an inconsistency on how segments are treated vs indexes and cache...